### ISSUE TITLE
scripts problems

### ISSUE THREAD

**Issue by gina-belmonte** [ellisdg/3DUnetCNN#312]
Hello, I have been using your excellent network. I am currently facing issues with the prediction script. The script calls the 'volumetric_prediction' function using filenames instead of a dataloader. I was wondering if there have been any updates regarding this matter.

**Comment by ellisdg** [comment]
@gina-belmonte, thank you! I'll incorporate those changes.
I was testing out the prediction last night, and I ran into an issue with NibabelWriter not resampling correctly. So I turned the resampling off, that fixed the one problem, but also means that the output images might not be aligned correctly with the input images. I'll try to look more into it and figure out a way around using the resampling in NibabelWriter.

**Comment by gina-belmonte** [comment]
I should have also made this change in unet3d/predict/volumetric.py, but I'm not sure what your original intention was with the --activation parameter in the predict function (perhaps the default should be sigmoid instead of None?):

'''
@@ -100,7 +99,7 @@ def volumetric_predictions(model, dataloader, prediction_dir, activation=None,
                 predictions = torch.sigmoid(predictions)
             elif activation == "softmax":
                 predictions = torch.softmax(predictions, dim=1)
-            else:
+            elif activation is not None:
                 predictions = getattr(torch, activation)(predictions)
             batch_size = x.shape[0]
             for batch_idx in range(batch_size):
'''

**Comment by ellisdg** [comment]
I was thinking that the default would be None so that it would simply output the logit values and the user could write a different to prostprocess the logit predictions. But I guess sigmoid might be the most common.

**Comment by ellisdg** [comment]
Oh I see. Thanks for helping me fix that. It should be fixed on the predict branch now.

**Comment by gina-belmonte** [comment]
Thanks, I see you changed resample.py too, I need to retrain the model?

**Comment by ellisdg** [comment]
No, nothing should be changed for model training.
Based on my tests, the predict.py is working now and writing the images to the correct output space. I'm going to go ahead and merge the pull request which will close this issue. Feel free to open the issue back up if you find the script isn't working for you.
